### PR TITLE
eclipse.ini: skip comment lines  #361

### PR DESCRIPTION
**Pull request by jukzi**
Commment lines have '#' as its first character

https://github.com/eclipse-equinox/p2/issues/361

### CODE REVIEW

**Comment by jukzi** [comment]
@merks mind to review?

**Comment by merks** [comment]
It looks like you're treating `!` as a comment character but the wiki link you showed mentions `;` and not `!`.

I was going to ask, could you please double check what the actual *.ini reader is doing at runtime?  But why ask someone else to do something you can do yourself.  😱  I believe this is the place where the native launcher ignores lines that start with '#'.

https://github.com/eclipse-equinox/equinox/blob/8beb9ce0b1dea1176b06ad7300f2663ff6db4eff/features/org.eclipse.equinox.executable.feature/library/eclipseConfig.c#L146-L150

This logic does not ignore any other from of content.  It also does not ignore lines where there is whitespace before the comment.  So I think it's best that you support only and exactly what will really work at runtime as a commented out line, i.e., a line starting with `#`.

**Comment by laeubi** [comment]
@jukzi du you like to complete this before freeze on Monday?

**Comment by jukzi** [comment]
thanks for the reminder, i changed commentlines to start with '#' only

**Review by merks — APPROVED** [review]
This looks exactly right.

File: bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java [hunk 1]
```diff
@@ -204,16 +204,20 @@ public static URI fromFileURL(String url) throws URISyntaxException {
...
 	public static List<String> loadFile(File file) throws IOException {
 		try (BufferedReader br = new BufferedReader(new FileReader(file));) {
 			String line;
 			List<String> list = new ArrayList<>();
 			while ((line = br.readLine()) != null) {
-				//skip whitespace
-				if (line.trim().length() > 0)
+				// skip whitespace and comments
+				String stripped = line.strip();
+				if (stripped.length() > 0 && !line.startsWith("#")) {//$NON-NLS-1$
```
Comment (HannesWell) [inline bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java:218]: Could be `!stripped.isEmpty()`

**Review by HannesWell — APPROVED** [review]
LGTM.
Just one suggestion below.

File: bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java (hunk 1 above)
Comment (jukzi) [reply bundles/org.eclipse.equinox.frameworkadmin.equinox/src/org/eclipse/equinox/internal/frameworkadmin/equinox/utils/FileUtils.java]: done, please submit

**Pull request merged**